opentreemap/exporter/tasks.py

```diff
@@ -204,15 +208,43 @@ def async_csv_export(job, model, query, display_filters):
 
 
 def _csv_field_header_map(field_names):
-    # Our query uses plot-centric field names but our csv wants tree-centric
-    # header names. So convert e.g. "tree__canopy_height" -> "canopy_height"
-    # and "width" -> "plot__width".
-    map = {}
-    for name in field_names:
-        if name.startswith('tree__'):
-            header = name[6:]
+    map = OrderedDict()
+    # TODO: make this conditional based on whether or not
+    # we are performing a complete export or an "importable" export
+    omit = {'readonly',
+            'tree__readonly',
+            'tree__species', 'tree__id',
+            'tree__species__fact_sheet_url',
+            'tree__species__fall_conspicuous',
+            'tree__species__flower_conspicuous',
+            'tree__species__flowering_period',
+            'tree__species__fruit_or_nut_period',
+            'tree__species__has_wildlife_value',
+            'tree__species__id',
+            'tree__species__is_native',
+            'tree__species__max_diameter',
+            'tree__species__max_height',
+            'tree__species__otm_code',
+            'tree__species__palatable_human',
+            'tree__species__plant_guide_url'}
+
+    field_names = field_names - omit
+
+    for name, header in fields.trees.EXPORTER_PAIRS:
+        if name in field_names:
+            map[name] = header
+            field_names.remove(name)
+
+    for name in sorted(field_names):
+        if name.startswith('udf:'):
+            header = 'Planting Site: ' + name[4:]
+        elif name.startswith('tree__udf:'):
+            header = 'Tree: ' + name[10:]
         else:
-            header = 'plot__%s' % name
+            # TODO: log that we have an uncaught field that may be
+            # making re-imports impossible. This is better than crashing
+            # an export that is otherwise usable.
+            continue
```

Comment:
Logging of "impossible" errors is always a good plan.  This diff should do it.
```diff
diff --git a/opentreemap/exporter/tasks.py b/opentreemap/exporter/tasks.py
index 9d97cbe..7b64205 100644
--- a/opentreemap/exporter/tasks.py
+++ b/opentreemap/exporter/tasks.py
@@ -4,6 +4,7 @@ from __future__ import unicode_literals
 from __future__ import division
 
 import csv
+import logging
 
 from contextlib import contextmanager
 from functools import wraps
@@ -23,14 +24,20 @@ from treemap.udf import UserDefinedCollectionValue
 from treemap.lib.object_caches import udf_defs
 
 from djqscsv import write_csv, generate_filename
-from exporter.models import ExportJob
 
+from opentreemap.util import add_rollbar_handler
+
+from exporter.models import ExportJob
 from exporter.user import write_users
 from exporter.util import sanitize_unicode_record
 
 from importer import fields
 
 
+logger = logging.getLogger(__name__)
+add_rollbar_handler(logger, level=logging.INFO)
+
+
 @contextmanager
 def _job_transaction_manager(job_pk):
     job = ExportJob.objects.get(pk=job_pk)
@@ -241,10 +248,8 @@ def _csv_field_header_map(field_names):
         elif name.startswith('tree__udf:'):
             header = 'Tree: ' + name[10:]
         else:
-            # TODO: log that we have an uncaught field that may be
-            # making re-imports impossible. This is better than crashing
-            # an export that is otherwise usable.
-            continue
+            logger.warn('Unrecognized export field name',
+                        extra={'extra_data': {'field_name': name}})
         map[name] = header
     return map
```